tree/dataframe/src/RDFInterfaceUtils.cxx

```diff
@@ -599,22 +735,20 @@ void BookFilterJit(const std::shared_ptr<RJittedFilter> &jittedFilter, void *pre
 {
    const auto &dsColumns = ds ? ds->GetColumnNames() : ColumnNames_t{};
 
-   // not const because `ColumnTypesAsStrings` might delete redundant matches and replace variable names
-   auto usedBranches = FindUsedColumnNames(expression, branches, customCols.GetNames(), dsColumns, aliasMap);
-   auto varNames = ReplaceDots(usedBranches);
-   auto dotlessExpr = std::string(expression);
-   const auto usedColTypes =
-      ColumnTypesAsString(usedBranches, varNames, aliasMap, tree, ds, dotlessExpr, customCols);
+   const auto parsedExpr =
+      ParseRDFExpression(std::string(expression), branches, customCols.GetNames(), dsColumns, aliasMap);
+   const auto exprVarTypes = GetColumnTypes(parsedExpr, tree, ds, customCols);
 
    TRegexp re("[^a-zA-Z0-9_]?return[^a-zA-Z0-9_]");
```

Comment:
Didn't think about this much. What puzzles me, though, is that you want something that's not something before return, but that's optional. In the end, you don't care if there's anything before `return`.

Reply:
Note that I meant the line the github *doesn't show* here.

Reply:
we care if there is something before `return` that _is_ `a-zA-Z0-9_`: we don't want to match something like `"var_return"`

Reply:
That's what I meant:
```
TPRegexp re2("[^a-zA-Z0-9_]?return")
root [4] re2.Match("areturn")
(int) 1
```

Reply:
I see! Well that's a bug I introduced in February 2019 then :+1: Thanks for spotting it! I'll fix it in another PR if that's ok

Reply:
Fixed in #5487